src/redux/news/news.operations.js

```diff
@@ -59,99 +17,170 @@ const getArticleById = (id) =>
-            message {
+            title {
               lang
               value
             }
-            statusCode
           }
         }
-      }
-    `,
-    fetchPolicy: 'no-cache'
-  });
+      `
+    });
+    const { data } = result;
+    return data.getAllNews;
+  } catch (e) {
+    return e;
+  }
+};
 
-const deleteArticle = async (id) => {
-  const result = await client.mutate({
-    variables: { id },
-    mutation: gql`
-      mutation($id: ID!) {
-        deleteNews(id: $id) {
-          ... on News {
-            author {
-              name {
+const getArticleById = async (id) => {
+  try {
+    const result = await client.query({
+      variables: { id },
+      query: gql`
+        query($id: ID!) {
+          getNewsById(id: $id) {
+            ... on News {
+              __typename
+              title {
+                lang
+                value
+              }
+              text {
+                lang
                 value
               }
+              images {
+                primary {
+                  medium
+                }
+                additional {
+                  large
+                }
+              }
+              video
+              author {
+                name {
+                  lang
+                  value
+                }
+                image {
+                  small
+                }
+              }
+              date
             }
-          }
-          ... on Error {
-            message {
-              lang
+            ... on Error {
+              __typename
+              message {
+                lang
+                value
+              }
+              statusCode
             }
-            statusCode
           }
         }
-      }
-    `,
-    fetchPolicy: 'no-cache'
-  });
-  client.resetStore();
-  return result;
+      `,
+      fetchPolicy: 'no-cache'
+    });
+    const { data } = result;
+    return data.getNewsById;
+  } catch (e) {
+    return e;
+  }
 };
 
-const createArticle = async (news) => {
-  const result = await client.mutate({
-    mutation: gql`
-      mutation($news: NewsInput!) {
-        addNews(news: $news) {
-          author {
-            name {
-              value
+const deleteArticle = async (id) => {
+  try {
+    const result = await client.mutate({
+      variables: { id },
+      mutation: gql`
+        mutation($id: ID!) {
+          deleteNews(id: $id) {
+            ... on News {
+              author {
+                name {
+                  value
+                }
+              }
+            }
+            ... on Error {
+              message {
+                lang
+              }
+              statusCode
             }
           }
         }
-      }
-    `,
-    variables: { news }
-  });
-  client.resetStore();
-  return result;
+      `,
+      fetchPolicy: 'no-cache'
+    });
+    client.resetStore();
+    const { data } = result;
+    return data.deleteNews;
+  } catch (e) {
+    return e;
```

Comment:
Try to implement it in another way, you have to kinds of errors:
- `ServerError` that is handled inside `catch` block in the saga
- a custom error that is sent inside message property

First one is going to be automatically thrown and goes to the `catch` block, but the second one should be thrown manually after getting result from the server
`if(res.data.deleteNews.message) { throw new SomeCustomError(...) }`.

`try {...} catch {..}` block inside operation is not needed anymore

Reply:
Also, please, never return an error, error should be thrown as quickly as possible to be handled in some `catch` block

Reply:
Changed!